operator/v1/types.go

```diff
@@ -126,6 +127,7 @@ type OperatorStatus struct {
 	ReadyReplicas int32 `json:"readyReplicas"`
 
 	// generations are used to determine when an item needs to be reconciled or has changed in a way that needs a reaction.
+	// +listType=atomic
```

Comment:
Should be a map
```suggestion
	// +listType=map
	// +listMapKey=group
	// +listMapKey=resource
	// +listMapKey=namespace
	// +listMapKey=name
```

Reply:
We can't use namespace as a map key because it's not a required field. You'll need to not include that one here

Reply:
I did have to add some required field tags for some of the other list "maps", so I assume that's okay other than for this namespace one?

Reply:
Can you give me a list of which fields you had to make required? I don't think we can actually. So we may have to move back to atomic, give me some time and I'll confirm before you go back and forth changing and changing back 😓

Reply:
Sure, the following lists asked me to add `+kubebuilder:validation:Required` tags. These internal fields did not have any tags on them:
- `GenerationStatus` - group, resource, name
- `OperatorCondition` - type
- `NodeStatus` - nodeName